ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -59,11 +63,17 @@ targetPeers :: (MonadSTM m, Ord peeraddr)
 targetPeers PeerSelectionActions{readPeerSelectionTargets}
             st@PeerSelectionState{
               localRootPeers,
-              targets
+              targets,
+              currentLedgerStateJudgement
             } =
     Guarded Nothing $ do
       targets' <- readPeerSelectionTargets
-      check (targets' /= targets && sanePeerSelectionTargets targets')
+      -- If the node is in fallback state we shouldn't touch the targets until
+      -- we are caught up again
+      check (  currentLedgerStateJudgement /= TooOld
```

Comment:
```suggestion
check (not hasOnlyBootstrapPeers)
```

Make a note regarding churn being idempotent (its still working but we don't notice it) while in fallback state

Reply:
This suggestion isn't right. If the targets change when in `hasOnlyBootstrapPeers` state then the local roots can diminish and get removed, leaving non trusted nodes in the known set

Reply:
That's right we have to also look at the bootstrap peers enabler flag.